lib/tasks/porfolios.rake

```diff
@@ -0,0 +1,39 @@
+require 'rake'
+
+namespace :portfolio do
+  desc "Remove RBAC roles for discarded portfolios"
+  task :remove_discarded_rbac => :environment do
+    raise "Please provide a user yaml file" unless ENV['USER_FILE']
+    request = create_request(ENV['USER_FILE'])
+    ManageIQ::API::Common::Request.with_request(request) do
+      roles = []
+      opts = { limit: 500, name: "catalog-portfolios" }
+
+      RBAC::Service.call(RBACApiClient::RoleApi) do |api|
+        roles = RBAC::Service.paginate(api, :list_roles, opts).to_a
+      end
+
+      Portfolio.with_discarded.each do |portfolio|
```

Comment:
Cloud this be something like: 
```ruby
PortfolioItem.with_discarded.discarded.where(:tenant_id => <org_id_from_user_file>)
```

Then you do not need the `next unless portfolio.discarded?` check and we would only process Portfolios for the defined tenant.